chromium_src/url/origin.h

```diff
@@ -0,0 +1,21 @@
+/* Copyright 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_URL_ORIGIN_H_
+#define BRAVE_CHROMIUM_SRC_URL_ORIGIN_H_
+
+#define GetTupleOrPrecursorTupleIfOpaque                                   \
+  NotUsed() const;                                                         \
+  /* Checks if origin is opaque and nonce is initialized. */               \
+  bool CanUseNonceForEphemeralStorageKeying() const;                       \
```

Comment:
I don't think this method needs to be on Origin, does it? It doesn't seem right to me to attach ephemeral storage specific logic to Origin and we have access to both `opaque()` and now the raw_token through `GetNonceForEphemeralStorageKeying`. Related I think we should change `GetNonceForEphemeralStorageKeying` -> `raw_token()` or something. Maybe both methods can become utility functions that take Origin as a param?

Reply:
agree, it doesn't have to be here. Moved it into a dedicated helper to access Origin internals related to Ephemeral Storage more explicitly.